src/services/export.service.ts

```diff
@@ -191,41 +219,53 @@ export class ExportService {
     this.adjustColumnWidth(worksheet);
   }
 
-  private getPopulations = (testCaseExcelExportDto: TestCaseExcelExportDto) => {
-    let populations: PopulationDto[] = [];
-    testCaseExcelExportDto.testCaseExecutionResults?.forEach(
-      (result: TestCaseExecutionResultDto) => {
-        if (result.populations?.length > 0) {
-          populations = result.populations;
-        }
-      },
-    );
-    return populations;
-  };
-
-  private populateTestCaseExpectedAndActual(
+  private populatePopExpected(
     testCaseData,
-    result: TestCaseExecutionResultDto,
-    population: PopulationDto,
+    populationDto: PopulationDto,
+    result,
     index: number,
     failedIndexes: number[],
   ) {
     let foundPopulation: PopulationDto = null;
     result.populations?.forEach((currentPopulation) => {
-      if (currentPopulation.name === population.name) {
+      if (currentPopulation.name === populationDto.name) {
         foundPopulation = currentPopulation;
       }
     });
-    testCaseData.push(foundPopulation?.expected, foundPopulation?.actual);
     if (foundPopulation?.expected !== foundPopulation?.actual) {
       if (failedIndexes.indexOf(index) === -1) {
         //if not in the array
         failedIndexes.push(index);
       }
     }
-    return foundPopulation;
+    testCaseData.push(foundPopulation?.expected);
+  }
+  private populatePopActual(
+    testCaseData,
+    populationDto: PopulationDto,
+    result,
+  ) {
+    let foundPopulation: PopulationDto = null;
+    result.populations?.forEach((currentPopulation) => {
+      if (currentPopulation.name === populationDto.name) {
+        foundPopulation = currentPopulation;
+      }
+    });
+    testCaseData.push(foundPopulation?.actual);
   }
 
+  private getPopulations = (testCaseExcelExportDto: TestCaseExcelExportDto) => {
+    let populations: PopulationDto[] = [];
+    testCaseExcelExportDto.testCaseExecutionResults?.forEach(
+      (result: TestCaseExecutionResultDto) => {
+        if (result.populations?.length > 0) {
+          populations = result.populations;
+        }
+      },
+    );
+    return populations;
```

Comment:
I'm lil bit confused about this loop. Are you trying to get populations from any one of the executionResults? because this loop will iterate even after you fulfill `result.populations?.length > 0`. there is no breaking once you get the populations and it keeps looping and overriding `populations = result.populations;`. if you just want populations from any execution result, then:
```suggestion
    const result  = testCaseExcelExportDto.testCaseExecutionResults?.find(result) => result.populations?.length > 0);
    return result? result.populations : [];
```

Reply:
@adongare Changed